### DISCUSSION TITLE
Fortinet syslog collection stops after a short time.

### DISCUSSION THREAD

**Discussion by Rbartram** [Security-Onion-Solutions/securityonion#5434] [Unsupported Versions] [answered]
Hello I've configured filebeat to capture syslog data from our Fortigate firewall.  It works for a few minutes and then stops.  

'tcpdump -i ens192 port 9004' shows packets continuously hitting the server interface
'tcpdump -i docker0 port 9004' shows packets hitting the docker0 interface, but they stop after 5 to 10 minutes.

Issuing 'firewall-cmd --reload' causes packets to resume on the docker0 interface, but only for another 5 to 10 minutes.

The sensor server was installed from the 2.3.70 ISO image release before Tuesday 8/31/2021.

Please advise where I might look for more information.

**Comment by dougburks** [comment]
Is it possible that Wazuh Active Response is blocking the IP address of your Fortigate firewall?

**Reply by Rbartram** [reply]
That's a good thought.  I checked wazuh's active-response.log and it contains only 4 old records from Sep 2 that don't appear to involve port 9004.

    Thu Sep  2 19:09:51 UTC 2021 /var/ossec/active-response/bin/host-deny.sh add - 10.1.1.1 1630609791.19142 5706
    Thu Sep  2 19:09:51 UTC 2021 /var/ossec/active-response/bin/firewall-drop.sh add - 10.1.1.1 1630609791.19142 5706
    Thu Sep  2 19:19:52 UTC 2021 /var/ossec/active-response/bin/host-deny.sh delete - 10.1.1.1 1630609791.19142 5706
    Thu Sep  2 19:19:52 UTC 2021 /var/ossec/active-response/bin/firewall-drop.sh delete - 10.1.1.1 1630609791.19142 5706

**Reply by Rbartram** [reply]
The messages log contain the following entries that correspond with syslog working/not working.


At 21:51:34, I ran firewall-cmd --reload
At 21:51:34, Packets started flowing to 172.17.0.4.9004
At 21:53:28, Packets stopped flowing to 172.17.0.4.9004
Packets continue to hit the physical interface of the host, just not reaching the docker interface.

messages
----------
    Sep  8 21:51:34 sof-01 firewalld[15069]: WARNING: COMMAND_FAILED: '/usr/sbin/iptables -w10 -D FORWARD -i docker0 -o docker0 -j DROP' failed: iptables: Bad rule (does a matching rule exist in that chain?).
    Sep  8 21:51:34 sof-01 dockerd: time="2021-09-08T21:51:34.858320283Z" level=info msg="Firewalld: interface docker0 already part of docker zone, returning"
    Sep  8 21:51:34 sof-01 dockerd: time="2021-09-08T21:51:34.924149144Z" level=info msg="Firewalld: interface docker0 already part of docker zone, returning"
    Sep  8 21:51:35 sof-01 firewalld[15069]: WARNING: COMMAND_FAILED: '/usr/sbin/iptables -w10 -D FORWARD -i docker0 -o docker0 -j DROP' failed: iptables: Bad rule (does a matching rule exist in that chain?).
    Sep  8 21:51:35 sof-01 dockerd: time="2021-09-08T21:51:35.193448523Z" level=info msg="Firewalld: interface docker0 already part of docker zone, returning"
    Sep  8 21:51:35 sof-01 dockerd: time="2021-09-08T21:51:35.262742471Z" level=info msg="Firewalld: interface docker0 already part of docker zone, returning"
    Sep  8 21:52:01 sof-01 systemd: Created slice User Slice of root.
    Sep  8 21:52:01 sof-01 systemd: Started Session 19510 of user root.
    Sep  8 21:52:01 sof-01 systemd: Started Session 19511 of user root.
    Sep  8 21:52:04 sof-01 systemd: Removed slice User Slice of root.
    Sep  8 21:53:01 sof-01 systemd: Created slice User Slice of root.
    Sep  8 21:53:01 sof-01 systemd: Started Session 19512 of user root.
    Sep  8 21:53:01 sof-01 systemd: Started Session 19513 of user root.
    Sep  8 21:53:04 sof-01 systemd: Removed slice User Slice of root.
    Sep  8 21:53:12 sof-01 systemd: Started /usr/bin/yum --quiet --assumeyes clean expire-cache.
    Sep  8 21:53:12 sof-01 systemd: Started /usr/bin/yum --quiet --assumeyes check-update --setopt=autocheck_running_kernel=false.
    Sep  8 21:53:14 sof-01 systemd: Started /usr/bin/yum --quiet list available openssl openssh-server bash.
    Sep  8 21:53:15 sof-01 dbus[722]: [system] Activating via systemd: service name='org.freedesktop.timedate1' unit='dbus-org.freedesktop.timedate1.service'
    Sep  8 21:53:15 sof-01 systemd: Starting Time & Date Service...
    Sep  8 21:53:15 sof-01 dbus[722]: [system] Successfully activated service 'org.freedesktop.timedate1'
    Sep  8 21:53:15 sof-01 systemd: Started Time & Date Service.
    Sep  8 21:53:28 sof-01 kernel: IPTables-dropped: IN=ens192 OUT=docker0 MAC=00:50:56:a3:f7:aa:d4:76:a0:6b:22:01:08:00 SRC=10.1.1.1 DST=172.17.0.4 LEN=936 TOS=0x00 PREC=0x00 TTL=63 ID=22132 PROTO=UDP SPT=10437 DPT=9004 LEN=916

**Comment by Rbartram** [comment]
I just discovered that packets stop flowing at the same time /etc/sysconfig/iptables is updated.  It seems that the sensor server pulls a config update from the manager at the same time.  Any ideas on how I resolve this? 

This is the output of iptables -S when packets are flowing (after I run firewall-cmd --reload)
--------
    -P INPUT ACCEPT
    -P FORWARD DROP
    -P OUTPUT ACCEPT
    -N DOCKER
    -N DOCKER-ISOLATION-STAGE-1
    -N DOCKER-ISOLATION-STAGE-2
    -N DOCKER-USER
    -N FORWARD_IN_ZONES
    -N FORWARD_OUT_ZONES
    -N FORWARD_direct
    -N FWDI_docker
    -N FWDI_docker_allow
    -N FWDI_docker_deny
    -N FWDI_docker_log
    -N FWDI_public
    -N FWDI_public_allow
    -N FWDI_public_deny
    -N FWDI_public_log
    -N FWDO_docker
    -N FWDO_docker_allow
    -N FWDO_docker_deny
    -N FWDO_docker_log
    -N FWDO_public
    -N FWDO_public_allow
    -N FWDO_public_deny
    -N FWDO_public_log
    -N INPUT_ZONES
    -N INPUT_direct
    -N IN_docker
    -N IN_docker_allow
    -N IN_docker_deny
    -N IN_docker_log
    -N IN_public
    -N IN_public_allow
    -N IN_public_deny
    -N IN_public_log
    -N OUTPUT_direct
    -A INPUT -m conntrack --ctstate RELATED,ESTABLISHED -j ACCEPT
    -A INPUT -i lo -j ACCEPT
    -A INPUT -j INPUT_direct
    -A INPUT -j INPUT_ZONES
    -A INPUT -m conntrack --ctstate INVALID -j LOG --log-prefix "STATE_INVALID_DROP: "
    -A INPUT -m conntrack --ctstate INVALID -j DROP
    -A INPUT -j LOG --log-prefix "FINAL_REJECT: "
    -A INPUT -j REJECT --reject-with icmp-host-prohibited
    -A FORWARD -j DOCKER-ISOLATION-STAGE-1
    -A FORWARD -j DOCKER-USER
    -A FORWARD -o docker0 -m conntrack --ctstate RELATED,ESTABLISHED -j ACCEPT
    -A FORWARD -o docker0 -j DOCKER
    -A FORWARD -i docker0 ! -o docker0 -j ACCEPT
    -A FORWARD -i docker0 -o docker0 -j ACCEPT
    -A FORWARD -m conntrack --ctstate RELATED,ESTABLISHED -j ACCEPT
    -A FORWARD -i lo -j ACCEPT
    -A FORWARD -j FORWARD_direct
    -A FORWARD -j FORWARD_IN_ZONES
    -A FORWARD -j FORWARD_OUT_ZONES
    -A FORWARD -m conntrack --ctstate INVALID -j LOG --log-prefix "STATE_INVALID_DROP: "
    -A FORWARD -m conntrack --ctstate INVALID -j DROP
    -A FORWARD -j LOG --log-prefix "FINAL_REJECT: "
    -A FORWARD -j REJECT --reject-with icmp-host-prohibited
    -A OUTPUT -o lo -j ACCEPT
    -A OUTPUT -j OUTPUT_direct
    -A DOCKER -d 172.17.0.3/32 ! -i docker0 -o docker0     -P tcp -m tcp --dport 1514 -j ACCEPT
    -A DOCKER -d 172.17.0.4/32 ! -i docker0 -o docker0     -P tcp -m tcp --dport 9004 -j ACCEPT
    -A DOCKER -d 172.17.0.4/32 ! -i docker0 -o docker0     -P udp -m udp --dport 9004 -j ACCEPT
    -A DOCKER -d 172.17.0.4/32 ! -i docker0 -o docker0     -P tcp -m tcp --dport 5066 -j ACCEPT
    -A DOCKER -d 172.17.0.4/32 ! -i docker0 -o docker0     -P udp -m udp --dport 514 -j ACCEPT
    -A DOCKER -d 172.17.0.2/32 ! -i docker0 -o docker0     -P tcp -m tcp --dport 80 -j ACCEPT
    -A DOCKER -d 172.17.0.3/32 ! -i docker0 -o docker0     -P tcp -m tcp --dport 1515 -j ACCEPT
    -A DOCKER -d 172.17.0.3/32 ! -i docker0 -o docker0     -P udp -m udp --dport 1514 -j ACCEPT
    -A DOCKER -d 172.17.0.4/32 ! -i docker0 -o docker0     -P tcp -m tcp --dport 514 -j ACCEPT
    -A DOCKER -d 172.17.0.2/32 ! -i docker0 -o docker0     -P tcp -m tcp --dport 443 -j ACCEPT
    -A DOCKER -d 172.17.0.3/32 ! -i docker0 -o docker0     -P tcp -m tcp --dport 55000 -j ACCEPT
    -A DOCKER-ISOLATION-STAGE-1 -j RETURN
    -A DOCKER-ISOLATION-STAGE-2 -j RETURN
    -A DOCKER-USER -j RETURN
    -A FORWARD_IN_ZONES -i bond0 -g FWDI_public
    -A FORWARD_IN_ZONES -i ens224 -g FWDI_public
    -A FORWARD_IN_ZONES -i docker0 -g FWDI_docker
    -A FORWARD_IN_ZONES -g FWDI_public
    -A FORWARD_OUT_ZONES -o bond0 -g FWDO_public
    -A FORWARD_OUT_ZONES -o ens224 -g FWDO_public
    -A FORWARD_OUT_ZONES -o docker0 -g FWDO_docker
    -A FORWARD_OUT_ZONES -g FWDO_public
    -A FWDI_docker -j FWDI_docker_log
    -A FWDI_docker -j FWDI_docker_deny
    -A FWDI_docker -j FWDI_docker_allow
    -A FWDI_docker -j ACCEPT
    -A FWDI_public -j FWDI_public_log
    -A FWDI_public -j FWDI_public_deny
    -A FWDI_public -j FWDI_public_allow
    -A FWDI_public     -P icmp -j ACCEPT
    -A FWDO_docker -j FWDO_docker_log
    -A FWDO_docker -j FWDO_docker_deny
    -A FWDO_docker -j FWDO_docker_allow
    -A FWDO_docker -j ACCEPT
    -A FWDO_public -j FWDO_public_log
    -A FWDO_public -j FWDO_public_deny
    -A FWDO_public -j FWDO_public_allow
    -A INPUT_ZONES -i bond0 -g IN_public
    -A INPUT_ZONES -i ens224 -g IN_public
    -A INPUT_ZONES -i docker0 -g IN_docker
    -A INPUT_ZONES -g IN_public
    -A IN_docker -j IN_docker_log
    -A IN_docker -j IN_docker_deny
    -A IN_docker -j IN_docker_allow
    -A IN_docker -j ACCEPT
    -A IN_public -j IN_public_log
    -A IN_public -j IN_public_deny
    -A IN_public -j IN_public_allow
    -A IN_public     -P icmp -j ACCEPT
    -A IN_public_allow     -P tcp -m tcp --dport 22 -m conntrack --ctstate NEW,UNTRACKED -j ACCEPT
    -A IN_public_allow     -P tcp -m tcp --dport 22 -m conntrack --ctstate NEW,UNTRACKED -j ACCEPT

This is the output of iptables -S when packets are NOT flowing (after the sensor server pulls config/salt updates from the manager)
--------
    -P INPUT ACCEPT
    -P FORWARD DROP
    -P OUTPUT ACCEPT
    -N DOCKER
    -N DOCKER-ISOLATION-STAGE-1
    -N DOCKER-ISOLATION-STAGE-2
    -N DOCKER-USER
    -N FORWARD_IN_ZONES
    -N FORWARD_OUT_ZONES
    -N FORWARD_direct
    -N FWDI_docker
    -N FWDI_docker_allow
    -N FWDI_docker_deny
    -N FWDI_docker_log
    -N FWDI_public
    -N FWDI_public_allow
    -N FWDI_public_deny
    -N FWDI_public_log
    -N FWDO_docker
    -N FWDO_docker_allow
    -N FWDO_docker_deny
    -N FWDO_docker_log
    -N FWDO_public
    -N FWDO_public_allow
    -N FWDO_public_deny
    -N FWDO_public_log
    -N INPUT_ZONES
    -N INPUT_direct
    -N IN_docker
    -N IN_docker_allow
    -N IN_docker_deny
    -N IN_docker_log
    -N IN_public
    -N IN_public_allow
    -N IN_public_deny
    -N IN_public_log
    -N LOGGING
    -N OUTPUT_direct
    -A INPUT -s 127.0.0.1/32     -P udp -m udp -j ACCEPT
    -A INPUT -s 127.0.0.1/32     -P tcp -m tcp -j ACCEPT
    -A INPUT -s 172.17.0.0/24     -P udp -m udp -j ACCEPT
    -A INPUT -s 172.17.0.0/24     -P tcp -m tcp -j ACCEPT
    -A INPUT     -P tcp -m tcp --dport 22 -j ACCEPT
    -A INPUT -m conntrack --ctstate RELATED,ESTABLISHED -j ACCEPT
    -A INPUT -i lo -j ACCEPT
    -A INPUT -j INPUT_direct
    -A INPUT -j INPUT_ZONES
    -A INPUT -m conntrack --ctstate INVALID -j LOG --log-prefix "STATE_INVALID_DROP: "
    -A INPUT -m conntrack --ctstate INVALID -j DROP
    -A INPUT -j LOG --log-prefix "FINAL_REJECT: "
    -A INPUT -j REJECT --reject-with icmp-host-prohibited
    -A INPUT     -P icmp -j ACCEPT
    -A INPUT -j LOGGING
    -A FORWARD -j DOCKER-ISOLATION-STAGE-1
    -A FORWARD -j DOCKER-USER
    -A FORWARD -o docker0 -m conntrack --ctstate RELATED,ESTABLISHED -j ACCEPT
    -A FORWARD -o docker0 -j DOCKER
    -A FORWARD -i docker0 ! -o docker0 -j ACCEPT
    -A FORWARD -i docker0 -o docker0 -j ACCEPT
    -A FORWARD -m conntrack --ctstate RELATED,ESTABLISHED -j ACCEPT
    -A FORWARD -i lo -j ACCEPT
    -A FORWARD -j FORWARD_direct
    -A FORWARD -j FORWARD_IN_ZONES
    -A FORWARD -j FORWARD_OUT_ZONES
    -A FORWARD -m conntrack --ctstate INVALID -j LOG --log-prefix "STATE_INVALID_DROP: "
    -A FORWARD -m conntrack --ctstate INVALID -j DROP
    -A FORWARD -j LOG --log-prefix "FINAL_REJECT: "
    -A FORWARD -j REJECT --reject-with icmp-host-prohibited
    -A OUTPUT -o lo -j ACCEPT
    -A OUTPUT -j OUTPUT_direct
    -A OUTPUT     -P icmp -m icmp --icmp-type 14 -j DROP
    -A DOCKER -d 172.17.0.2/32 ! -i docker0 -o docker0     -P tcp -m tcp --dport 443 -j ACCEPT
    -A DOCKER -d 172.17.0.3/32 ! -i docker0 -o docker0     -P tcp -m tcp --dport 55000 -j ACCEPT
    -A DOCKER -d 172.17.0.3/32 ! -i docker0 -o docker0     -P udp -m udp --dport 1514 -j ACCEPT
    -A DOCKER -d 172.17.0.4/32 ! -i docker0 -o docker0     -P tcp -m tcp --dport 514 -j ACCEPT
    -A DOCKER -d 172.17.0.4/32 ! -i docker0 -o docker0     -P tcp -m tcp --dport 5066 -j ACCEPT
    -A DOCKER -d 172.17.0.4/32 ! -i docker0 -o docker0     -P udp -m udp --dport 514 -j ACCEPT
    -A DOCKER -d 172.17.0.2/32 ! -i docker0 -o docker0     -P tcp -m tcp --dport 80 -j ACCEPT
    -A DOCKER -d 172.17.0.3/32 ! -i docker0 -o docker0     -P tcp -m tcp --dport 1515 -j ACCEPT
    -A DOCKER -d 172.17.0.3/32 ! -i docker0 -o docker0     -P tcp -m tcp --dport 1514 -j ACCEPT
    -A DOCKER -d 172.17.0.4/32 ! -i docker0 -o docker0     -P tcp -m tcp --dport 9004 -j ACCEPT
    -A DOCKER -d 172.17.0.4/32 ! -i docker0 -o docker0     -P udp -m udp --dport 9004 -j ACCEPT
    -A DOCKER-ISOLATION-STAGE-1 -j RETURN
    -A DOCKER-ISOLATION-STAGE-2 -j RETURN
    -A DOCKER-USER ! -i docker0 -o docker0 -m conntrack --ctstate RELATED,ESTABLISHED -j ACCEPT
    -A DOCKER-USER -s 10.17.100.10/32     -P udp -m udp --dport 514 -j ACCEPT
    -A DOCKER-USER -s 10.17.100.10/32     -P tcp -m tcp --dport 514 -j ACCEPT
    -A DOCKER-USER ! -i docker0 -o docker0 -j LOGGING
    -A DOCKER-USER -j RETURN
    -A FORWARD_IN_ZONES -i bond0 -g FWDI_public
    -A FORWARD_IN_ZONES -i ens224 -g FWDI_public
    -A FORWARD_IN_ZONES -i docker0 -g FWDI_docker
    -A FORWARD_IN_ZONES -g FWDI_public
    -A FORWARD_OUT_ZONES -o bond0 -g FWDO_public
    -A FORWARD_OUT_ZONES -o ens224 -g FWDO_public
    -A FORWARD_OUT_ZONES -o docker0 -g FWDO_docker
    -A FORWARD_OUT_ZONES -g FWDO_public
    -A FWDI_docker -j FWDI_docker_log
    -A FWDI_docker -j FWDI_docker_deny
    -A FWDI_docker -j FWDI_docker_allow
    -A FWDI_docker -j ACCEPT
    -A FWDI_public -j FWDI_public_log
    -A FWDI_public -j FWDI_public_deny
    -A FWDI_public -j FWDI_public_allow
    -A FWDI_public     -P icmp -j ACCEPT
    -A FWDO_docker -j FWDO_docker_log
    -A FWDO_docker -j FWDO_docker_deny
    -A FWDO_docker -j FWDO_docker_allow
    -A FWDO_docker -j ACCEPT
    -A FWDO_public -j FWDO_public_log
    -A FWDO_public -j FWDO_public_deny
    -A FWDO_public -j FWDO_public_allow
    -A INPUT_ZONES -i bond0 -g IN_public
    -A INPUT_ZONES -i ens224 -g IN_public
    -A INPUT_ZONES -i docker0 -g IN_docker
    -A INPUT_ZONES -g IN_public
    -A IN_docker -j IN_docker_log
    -A IN_docker -j IN_docker_deny
    -A IN_docker -j IN_docker_allow
    -A IN_docker -j ACCEPT
    -A IN_public -j IN_public_log
    -A IN_public -j IN_public_deny
    -A IN_public -j IN_public_allow
    -A IN_public     -P icmp -j ACCEPT
    -A IN_public_allow     -P tcp -m tcp --dport 22 -m conntrack --ctstate NEW,UNTRACKED -j ACCEPT
    -A IN_public_allow     -P tcp -m tcp --dport 22 -m conntrack --ctstate NEW,UNTRACKED -j ACCEPT
    -A LOGGING -m limit --limit 2/min -j LOG --log-prefix "IPTables-dropped: "
    -A LOGGING -j DROP

**Comment by dougburks** [comment, marked as answer]
Did you update firewall rules as described in the documentation?
https://docs.securityonion.net/en/2.3/firewall.html

**Reply by Rbartram** [reply]
Thank you!  I figured there was something that I still needed to learn...

**Reply by Rbartram** [reply]
From the manager server I did the following:

1. created a hostgroup called fortinet

       sudo so-firewall addhostgroup fortinet

2. added my firewall to the group

       sudo so-firewall includehost fortinet 10.1.1.1
3. created a portgroup called forinet

        sudo so-firewall addportgroup fortinet
4. added tcp port 9004 to the port group

        sudo so-firewall addport fortinet tcp 9004
5. added the following to /opt/so/saltstack/local/pillar/minions/sof-01_sensor.sls

        firewall:
          assigned_hostgroups:
            chain:
              DOCKER-USER:
                hostgroups:
                 fortinet:
                    portgroups:
                      - portgroups.fortinet
6. forced an update of the firewall config on the sensor server

        sudo salt sof-01_sensor state.apply firewall

Logs are flowing now!  Thanks again Doug.
